Summary

jgozal opened the pull request:
![image](https://user-images.githubusercontent.com/9560798/64913713-d62d3b00-d6f9-11e9-926b-750a5d788fa8.png)

jgozal commented on src/utils/timeSeriesResultsByOption.js:
```diff
@@ -4,7 +4,7 @@ import { FV, PMT } from 'formulajs/lib/financial'
 const COMPOUND_FREQUENCY = 12
```
just replaced loanAmount with loanAmt to keep consistency in this file, no other changes.

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -50,6 +50,35 @@ option2.term = 30
+  li {
+    margin-bottom: 1rem;
+  }
+`
```
sug: As discussed, if you wanted to pull these out, styled components would be faster to cmd-f

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -251,6 +282,9 @@ class MainForm extends React.Component {
 
   render () {
+    const shorterOption = this.state.options[this.state.shorterOption]
+    const longerOption = this.state.options[this.state.longerOption]
```
🙌

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -519,19 +546,93 @@ class MainForm extends React.Component {
           </Card>
         </Col>
         <Col xs='8'>
+          <Summary className='p-4 mr-5'>
```
sug: Perhaps this could just be `Row`?

hdavidzhu replied:
~For the spacing, you can apply `no-gutters` to this row and it will align:~

https://getbootstrap.com/docs/4.1/layout/grid/#no-gutters

Ah, looking at the alignment issues, it seems that padding is the explanation not having padding is what's causing a misalignment

jgozal replied:
> sug: Perhaps this could just be `Row`?

it is Row! I just styled it and gave it a more intuitive name.  What exactly should I do about spacing? I'm not sure if I follow

hdavidzhu replied:
Ah, it's related to styled components we talked about. The `Summary` seemed to just be a parent that wraps more specific child styles. After playing w/ it, I think this could just be Row w/o any child CSS in it

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -50,6 +50,35 @@ option2.term = 30
 
+const Summary = styled(Row)`
+  b {
+    font-weight: 500;
```
Do our `b`s need to look different in the app?

If this is the preferred weight, can we overwrite all the `b`s?

jgozal replied:
I'm not sure if we should change all `b`s on the app. When simply using the `b` tag it looks too bold. I could take this out and make a styled component called `lb` or `LightBold` and use that instead if you prefer

hdavidzhu replied:
After some of my own testing, I do like `500` to be the global bold styling since it doesn't scream at the user 😄I'm not sure how to best do global styling though. Thoughts?

https://github.com/jgozal/moneycompass/pull/31/files#diff-13bae51262f519caebe5af0bd2fcb33cR56

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -50,6 +50,35 @@ option2.term = 30
+  }
+
+  ol {
+    font-size: 15px;
```
q: What do things look like w/o any custom font-size changes?

jgozal replied:
relatively disproportionate in my opinion. You can give it a try.

hdavidzhu replied:
After removing some custom font sizes, I think it looks still fine:

![image](https://user-images.githubusercontent.com/3756871/64924603-666f8c80-d79b-11e9-8bfb-a2d7a1bd1c9b.png)

Another thing I've seen people do is to reduce the blackness of the body font to reduce emphasis. Perhaps we can give that a try too?

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -519,19 +546,93 @@ class MainForm extends React.Component {
+              <ol>
+                <li>
+                  With the <b>{shorterOption.term} year mortgage</b> you pay{' '}
+                  <b>{formatMoney(shorterOption.pmt)}</b> monthly for{' '}
```
Looks like this may need a sign flip

jgozal replied:
yeah I was thinking about that. I'm wondering if it would be better to turn everything into positives and handle the signs only in calculations (as opposed to handling them in the UI). I think almost every number in the app should be positive, including the ones on the table. Negatives are more confusing to the average joe

hdavidzhu replied:
Negatives would be accurate in the accounting world, and when we do calculations internally, we probably still want them negative.

Keeping the positive seems like a presentation responsibility for the user, so I think we should flip the signs when we display

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -519,19 +546,93 @@ class MainForm extends React.Component {
+                </li>
+              </ol>
+            </Col>
+            <span className='border mr-5' />
```
sug: You could probably just use a `border-right` on the prior column

jgozal replied:
it might be harder to center it that way, did you give it a try?

hdavidzhu commented on src/components/main-form.js:
```diff
@@ -50,6 +50,35 @@ option2.term = 30
+
+  .col {
+    margin-left: -2rem;
+  }
```
Ah, something I'm seeing is that you may not these custom. It's hidden in the design, but the explanation also has padding, which is why things align:

![image](https://user-images.githubusercontent.com/3756871/64914750-7c823c00-d70c-11e9-8a3e-43bfb1264fcf.png)

jgozal replied:
not quite following your thought process here. Can you elaborate?

hdavidzhu replied:
Hm, since it's tricky to pair right now, here's how it may look in code:

https://github.com/jgozal/moneycompass/pull/31/files#diff-13bae51262f519caebe5af0bd2fcb33cR552

If the summary itself has padding, then we can follow standard row column padding for Bootstrap.

Please run the example locally if you want to see what I mean!

jgozal commented:
@hdavidzhu to confirm, given your recent changes in the tmp branch, would the only thing to change here be the negative/positive signs? Could you take a second quick review at this to make sure I'm addressing all issues here before I merge it?
